components/PlantDetail.js

```diff
@@ -23,7 +26,24 @@ export default function PlantDetail({
   plantDetail,
   isOwned,
   handleToggleOwnedPlants,
+  handleDeletePlant,
 }) {
+  const [confirmDelete, setConfirmDelete] = useState(false);
+  const router = useRouter();
+
+  function handleDelete() {
+    setConfirmDelete(true);
+  }
+
+  function handleCancel() {
+    setConfirmDelete(false);
+  }
+
+  function handleConfirm() {
+    handleDeletePlant(plantDetail.id);
+    setConfirmDelete(false);
```

Comment:
This line of code would not be necessary because the page is changed directly afterwards and the confirmation dialog would not have to be closed separately.

Reply:
Okay, thanks :)